webapp/src/Entity/Testcase.php

```diff
@@ -132,6 +132,13 @@ class Testcase
      */
     private ?Problem $problem;
 
+    /**
+     * @ORM\ManyToOne(targetEntity="TestcaseGroup", inversedBy="testcases")
+     * @ORM\JoinColumn(name="testcasegroupid", referencedColumnName="testcasegroupid", onDelete="CASCADE")
+     * @Serializer\Exclude()
+     */
+    private ?TestcaseGroup $testcase_group;
```

Comment:
Mozna bych se nebal udelat tento link jako povinny. Tedy aby kazdy testcase mel nejakou skupinu.
Pro aktualni problemy muzeme udelat migracni skript, ktery zalozi default skupninu, ktera bude mit v sobe vsechnytestcasy daneho problemu a tato skupina bude mit procena = 100%

Zaroven by se ale muselo na jinem miste v kodu zajistit, ze kdyz se nekde vytvari problem, musi se automaticky vytvorit i defaultni skupina, do ktere se budou prirazovat testcasy. (nebo pri vytvareni prvniho testcasu)

Kdyz to ale vytvoris uz pri vztvareni problemu, bude to vypadat mozna prijemneji z UX, ze kdyz pujdes do editace testcasu, uz je budes moci zakladat v nejake skupine.

Uznavam, ze to je celkem velky zasah do DJ, ale myslim, ze to kapku zajisti "stabilitu" systemu a hlavne to bude primocarejsi pro budouci upravy -> nebudes muset testovat, jestli dany problem ma nejake skupiny, atd.. proste zavest pravidlo, ze kazy problem/testcase ma nejakou skupinu

Reply:
Aha, bude to muset byt az po vytvoreni prvniho testcase,.. ja se totiz tady celou dobu snazim najit cestu, jak odstranit asociaci mezi testcasegrouop a problem a tohle by neslo bez ni.

Tzn pak v te sablone nechat treba -1 nebo 0 jako default group a pokud se pri savu techto testcasu objevi tato skupina, vytvorit prave nejakou defaultni.

Pokud uz nejake testcases budou existovat, "vyhledat" vsechny groupy (bude muset existovat alespon 1) a z tohoto seznamu udelat selecboxy.

Pro vytvareni (dalsich) a editaci testcase groups vyuzit klidne oddelenou stranku mimo tu editaci s testcases

Reply:
a co kdyz se ze skupiny odeberou vsechny testcases a presunou do jinych - jak se pozna, do ktereho problemu skupina patri?

Reply:
ono by asi davalo smysl mit propojeni problem -> skupina -> testcase, ale to by byl dost zasah do dj

Reply:
1) pokus bych se, aby tato situace nenastala. Tedy napriklad kdyz ukladas/editujes testcases, ukladat si i do nejakeho formularoveho atributu seznam skupin (idcka?). Pokud nejaka skupina nebude vyuzita pro prirazeni testcasu, smazat ji.
2) DJ ma config check, kde se treba zobrazuje, ze mas contest bez problemu, ze problem v contesu nema nastavenou barvu atd.. tak tam by se mohl pridat check na skupiny, ktere nemaji zadny testcase v sobe